### ISSUE TITLE
🐛 Bug Report: plugin-catalog-backend-module-aws missing dependency in 1.11.0

### ISSUE THREAD

**Issue by jgrigg** [backstage/backstage#16380]
### 📜 Description

Since upgrading to backstage `1.11.0` the `@backstage/plugin-catalog-backend-module-aws` module is missing an explicit required dependency `@backstage/plugin-kubernetes-common`. Presumably this was coming along for the ride via another package's deps or transitives.

### 👍 Expected behavior

backend starts correctly.

### 👎 Actual Behavior with Screenshots

```sh
Starting backstage with config app-config.yaml and app-config.local.yaml
node:internal/modules/cjs/loader:998
  throw err;
  ^

Error: Cannot find module '@backstage/plugin-kubernetes-common'
Require stack:
- /app/node_modules/@backstage/plugin-catalog-backend-module-aws/dist/index.cjs.js
- /app/packages/backend/dist/index.cjs.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module.Hook._require.Module.require (/app/node_modules/require-in-the-middle/index.js:81:25)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/app/node_modules/@backstage/plugin-catalog-backend-module-aws/dist/index.cjs.js:5:30)
    at Module._compile (node:internal/modules/cjs/loader:1165:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1219:10)
    at Module.load (node:internal/modules/cjs/loader:1043:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1067:19)
    at Module.Hook._require.Module.require (/app/node_modules/require-in-the-middle/index.js:101:39) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/node_modules/@backstage/plugin-catalog-backend-module-aws/dist/index.cjs.js',
    '/app/packages/backend/dist/index.cjs.js'
  ]
}
Backstage stopping
```

### 👟 Reproduction steps

- create a fresh app
- yarn add @backstage/plugin-catalog-backend-module-aws
- configure the aws module in the catalog.ts
- start the backend 

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by freben** [comment]
This is getting closed by https://github.com/backstage/backstage/pull/16413 which should be released today

**Comment by benjdlambert** [comment]
Closing as got released part of the bonus 1.11.1 release.

**Comment by Remyapraveen** [comment]
Hi, I am using the backstage version of backstage:  1.12.1. I am currently trying to deploy my app into AWS EC2 instance after an integration with Authentication Providers. The app is working well in the local environment, but getting this error when I am trying to deploy it in AWS environment. 

```
app_1     | node:internal/modules/cjs/loader:444
app_1     |       throw err;
app_1     |       ^
app_1     |
app_1     | Error: Cannot find module '/app/packages/backend/dist\index.cjs.js'. Please verify that the package.json has a valid "main" entry
app_1     |     at tryPackage (node:internal/modules/cjs/loader:436:19)
app_1     |     at Module._findPath (node:internal/modules/cjs/loader:678:18)
app_1     |     at resolveMainPath (node:internal/modules/run_main:19:25)
app_1     |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:75:24)
app_1     |     at node:internal/main/run_main_module:23:47 {
app_1     |   code: 'MODULE_NOT_FOUND',
app_1     |   path: '/app/packages/backend/package.json',
app_1     |   requestPath: '/app/packages/backend'
app_1     | }
app_1     |
app_1     | Node.js v18.15.0

````

My .dockerignore file looks as default with following elements.

```

.git
.yarn/cache
.yarn/install-state.gz
node_modules
packages/*/src
packages/*/node_modules
plugins
*.local.yaml
````

Also, I somehow feel the path convention mentioned in here itself is wrong, "'/app/packages/backend/dist\index.cjs.js'". Any suggestions on how to rectify this would be of great help. TIA.
